app/services/idv/steps/threat_metrix_step_helper.rb

```diff
@@ -47,23 +47,31 @@ def threatmetrix_iframe_url(session_id)
         )
       end
 
-      def log_irs_tmx_fraud_check_event(result)
+      def log_irs_tmx_fraud_check_event(result, user)
         return unless IdentityConfig.store.irs_attempt_api_track_tmx_fraud_check_event
         return unless FeatureManagement.proofing_device_profiling_collecting_enabled?
+
         success = result[:review_status] == 'pass'
 
-        if !success && (tmx_summary_reason_code = result.dig(
-          :response_body,
-          :tmx_summary_reason_code,
-        ))
-          failure_reason = {
-            tmx_summary_reason_code: tmx_summary_reason_code,
-          }
+        unless success
+          fraud_event = FraudEvent.create(
+            user: user,
+            irs_session_id: irs_attempts_api_session_id,
+            # I think the to_s here is only needed in badly-constructed tests?
+            login_session_id: Digest::SHA1.hexdigest(user&.unique_session_id.to_s),
```

Comment:
Let me see if I can fix the test to not need this. I don't think it's sensible to get here with a nil user.